src/Symfony/Component/Messenger/Middleware/WrapsMessageHandlingInTransaction.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Middleware;
+
+use Doctrine\Common\Persistence\ManagerRegistry;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\Messenger\MiddlewareInterface;
+
+/**
+ * Wraps all handlers in a single doctrine transaction.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class WrapsMessageHandlingInTransaction implements MiddlewareInterface
+{
+    private $managerRegistry;
+    private $entityManagerName;
+
+    public function __construct(ManagerRegistry $managerRegistry, string $entityManagerName)
+    {
+        $this->managerRegistry = $managerRegistry;
+        $this->entityManagerName = $entityManagerName;
+    }
+
+    public function handle($message, callable $next)
+    {
+        $entityManager = $this->managerRegistry->getManager($this->entityManagerName);
+
+        if (!$entityManager instanceof EntityManagerInterface) {
+            throw new \InvalidArgumentException(sprintf('The ObjectManager with name "%s" must be an instance of EntityManagerInterface', $this->entityManagerName));
```

Comment:
This exception seems unnecessary IMHO, if `$this->entityManagerName` exists the `$entityManager` should be always an `EntityManagerInterface` instance ([see](https://github.com/doctrine/common/blob/a61bfddd39b887265117806e81cc6ca8da242fe1/lib/Doctrine/Common/Persistence/AbstractManagerRegistry.php#L161-L163)), otherwise if it doesn't exists Doctrine library throws its own `\InvalidArgumentException` ([see](https://github.com/doctrine/DoctrineBundle/blob/9e79906183c26557018ca6242043a1d4db40c36c/DependencyInjection/DoctrineExtension.php#L509-L517)).

Reply:
Hm.. But the doc block says ObjectManager..

Reply:
Ah! I see, NVM, sorry for the noise.

Reply:
Do not be sorry. Thank you for challenging this =)